### ISSUE TITLE
Bazel slow and flaky at HEAD on Travis CI

### ISSUE THREAD

**Issue by jart** [bazelbuild/bazel#1494]
Something mysterious happened about 25 days ago with Bazel HEAD. It caused Closure Rules' Travis CI tests to go from taking 15 minutes to 45 minutes. 

<img width="423" alt="screen shot 2016-07-09 at 12 57 56 am" src="https://cloud.githubusercontent.com/assets/49262/16706181/58870172-4570-11e6-9de0-c3fe5627ad73.png">

Travis builds at HEAD are also very frequently failing, because Travis is kill 9'ing Bazel process. Probably due to exceeding memory.

Here's the build history:

https://travis-ci.org/bazelbuild/rules_closure/builds

**Comment by jart** [comment]
It's also worth mentioning that Closure Rules enabled testing Bazel at HEAD exactly 25 days ago. So whatever happened, must have happened somewhere between 0.3.0 and 25 days ago. (See my [.travis.yml](https://github.com/bazelbuild/rules_closure/blob/master/.travis.yml) file for reference.)

**Comment by damienmg** [comment]
I am not seeing > 15min build. Each 45min build is 3 runs of Bazel. That's a travis issue.

**Comment by kchodorow** [comment]
It looks like
https://github.com/bazelbuild/rules_closure/commit/401aa58cf446d2e242973d5ff88f6274ed5db234
changes the .travis.yml to run three different builds, instead of just
one.  It looks like the elapsed time is still just 15 minutes:


​Source: https://travis-ci.org/bazelbuild/rules_closure/builds/137545464

**Comment by damienmg** [comment]
Closing this issue since it is not from Bazel

**Comment by jart** [comment]
I'm sorry, I explained this incorrectly. The build itself takes like 10 minutes. What's happening is it appears to be using a lot more memory at head. So Travis ends up kill 9'ing one of the processes.

<img width="343" alt="screen shot 2016-07-11 at 6 51 30 am" src="https://cloud.githubusercontent.com/assets/49262/16728378/f0f0bf9c-4733-11e6-84dc-69e101a13502.png">

Did something change in the way that Bazel schedules processes? Maybe Bazel is exceeding the resource limits I gave it?
